test/e2e/pages/user/user_billing_page.py

```diff
@@ -0,0 +1,120 @@
+from playwright.sync_api import Page, expect
+
+from ..base_class import BasePage
+from ..common.side_nav_bar import SideNavBar
+from ...data.models import TransactionData
+from ...utils.datetime_utils import compare_datetimes_with_tolerance, extract_datetime_from_text, parse_datetime
+
+class UserBillingPage(BasePage):
+    path: str = "/user/{username}/billing"
+
+    def __init__(self, page: Page, username: str):
+        super().__init__(page, username=username)
+
+        self.side_nav = SideNavBar(page)
+        self.tab_navigation = {
+            "overview": page.get_by_test_id("user-billing-tab-button-OverviewMenu"),
+            "history": page.get_by_test_id("user-billing-tab-button-HistoryMenu"),
+            "top_up": page.get_by_test_id("user-billing-tab-button-TopUpMenu"),
+        }
+
+        self.overview_menu = {
+            # Billing Overview
+            "estimated_due": page.get_by_test_id("overview-menu-estimated-due"),
+            "next_billing_date": page.get_by_test_id("overview-menu-next-billing-date"),
+            "active_subscriptions": page.get_by_test_id("overview-menu-active-subscriptions"),
+            "total_paid_cycle": page.get_by_test_id("overview-menu-total-paid-cycle"),
+            # Payment Method
+            "current_balance": page.get_by_test_id("overview-menu-current-balance"),
+            "top_up_button": page.get_by_test_id("overview-menu-top-up-balance-button"),
+        }
+
+        self.history_menu = {
+            "history_table": page.get_by_test_id("history-menu-table"),
+            "get_row": lambda row_id: page.get_by_test_id(f"history-menu-table-row-{row_id}"),
+        }
+
+        self.top_up_menu = {
+            "current_balance": page.get_by_test_id("top-up-menu-current-balance"),
+            "top_up_input": page.get_by_test_id("top-up-menu-input-amount-input-field"),
+            "add_credit_button": page.get_by_test_id("top-up-menu-add-credit-button"),
+            "get_quick_selection": lambda amount: page.get_by_test_id(f"top-up-menu-quick-selection-{amount}-option-button"),
+        }
+
+    def navigate_to_tab(self, tab: str):
+        self.tab_navigation[tab].click()
+        
+    def estimated_due_amount_should_be(self, amount: str):
+        expect(self.overview_menu["estimated_due"]).to_have_text(amount)
+
+    def next_billing_date_should_be(self, date: str, tolerance_minutes: int = 2):
```

Comment:
can we check only for the date ? I think checking for minutes only to add some wiggle room is meaningless.

Reply:
noted.